--- OpenBLAS-0.3.24/cmake/cc.cmake	2023-12-20 09:44:56.000000000 +0800
+++ OpenBLAS-0.3.24/cmake/cc.cmake	2023-12-20 09:50:15.283053620 +0800
@@ -36,9 +36,9 @@
 
     if (LOONGARCH64)
       if (BINARY64)
-        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=lp64")
+        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=lp64d")
       else ()
-        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=lp32")
+        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=ilp32d")
       endif ()
       set(BINARY_DEFINED 1)
     endif ()
--- OpenBLAS-0.3.24/Makefile.system	2023-12-20 09:52:09.000000000 +0800
+++ OpenBLAS-0.3.24/Makefile.system	2023-12-20 10:00:34.239841995 +0800
@@ -940,7 +940,7 @@
 ifeq ($(ARCH), loongarch64)
 LA64_ABI=$(shell $(CC) -mabi=lp64d -c $(TOPDIR)/cpuid_loongarch64.c -o /dev/null > /dev/null 2> /dev/null && echo lp64d)
 ifneq ($(LA64_ABI), lp64d)
-LA64_ABI=lp64
+LA64_ABI=lp64d
 endif
 CCOMMON_OPT += -march=loongarch64 -mabi=$(LA64_ABI)
 FCOMMON_OPT += -march=loongarch64 -mabi=$(LA64_ABI)
--- OpenBLAS-0.3.24/cmake/fc.cmake	2023-12-20 10:04:35.168171539 +0800
+++ OpenBLAS-0.3.24/cmake/fc.cmake	2023-12-20 10:04:05.596130715 +0800
@@ -61,9 +61,9 @@
     endif ()
     if (LOONGARCH64)
       if (BINARY64)
-        set(FCOMMON_OPT "${FCOMMON_OPT} -mabi=lp64")
+        set(FCOMMON_OPT "${FCOMMON_OPT} -mabi=lp64d")
       else ()
-        set(FCOMMON_OPT "${FCOMMON_OPT} -mabi=lp32")
+        set(FCOMMON_OPT "${FCOMMON_OPT} -mabi=ilp32d")
       endif ()
     endif ()
     if (RISCV64)
